R/mclosest.R

```diff
@@ -0,0 +1,85 @@
+#'@title Extract closest values in a pairwise manner between two matrices
+#'
+#' @description
+#'
+#' `mclosest` functions calculate the closest rows in a pairwise manner.
+#' It returns the index of the closest row in 'table' for each row in 'x'. 
+#'
+#' @param x `numeric`  matrix or data frame representing the query data. each 
+#' row will be compared to 'table'. Need to have the same numbers of rows and in 
+#' the same order as 'table'.
+#'
+#' @param table `numeric` matrix or data frame containing the reference data to 
+#' be matched with each row of 'x'. Need to have the same numbers of rows and in 
+#' the same order as 'x'.
+#'
+#' @param ppm `numeric` representing a relative, value-specific 
+#' parts-per-million (PPM) tolerance that is added to tolerance (default is 0). 
+#'
+#' @param tolerance `numeric` accepted tolerance.
+#'
+#' @return `numeric` vector of indices indicating the closest row of 'table' for
```

Comment:
Actually, it's a `integer` vector :) . I would maybe also say that a `NA` is returned for rows in x for which the difference for all rows in `table` of at least one column is larger than tolerated (depending on `tolerance` and `ppm`.

Reply:
What do you think of "If no suitable match is found for a row in 'x' based on the  specified tolerance' and 'ppm', the corresponding index is set to 'NA'." ? is it too vague ?